lib/MusicBrainz/Server/Form/Collection.pm

```diff
@@ -102,18 +102,31 @@ sub validate_collaborators {
     my $is_valid = 1;
     for my $collaborator (@collaborators) {
         my $id_field = $collaborator->field('id');
+        my $id = $id_field->value;
         my $name_field = $collaborator->field('name');
-        if (defined $name_field->value && !(defined $id_field->value)) {
-            my $editor = $self->ctx->model('Editor')->get_by_name($name_field->value);
+        my $name = $name_field->value;
+
+        if (defined $id) {
+            my $count =
+                grep { $_->field('id')->value eq $id } @collaborators;
+            if ($count > 1) {
+                $id_field->add_error(
+                    l('Please add each editor only once.')
+                );
```

Comment:
Do you think it's necessary to show an error in this case?  We could also just uniquify the IDs without bothering the user.

Reply:
I think it's useful because it feels unlikely the user did this willingly - so maybe they just clicked the wrong editor, and then if we accept it without erroring they don't realize something went wrong. But maybe I'm overthinking it.

Reply:
Then maybe it should specify what the issue might be, like "“{collaborator}” was added more than once. Did you intend to select someone else?" I would certainly be confused why the server would bother to tell me this if I did it by accident.

Reply:
That seems like a good text, used it.